### PR TITLE
Contrib/BitSight SPM Enhacement

### PR DESCRIPTION
**Pull request by crestdatasystems**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
BitSight SPM content pack enhacement.

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Names of existing commands have been changed to better meet XSOAR standards.
       - Context keys and their primary keys have been updated/corrected.
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by MLainer1 — CHANGES_REQUESTED** [review]
Good work, looks great :)
There is no max_fetch parameter in the fetch_incidents command, consider adding it. 
Please make sure to look at the suggestions, and request a review when you are done

File: Packs/BitSight/Integrations/BitSightForSecurityPerformanceManagement/test_data/company_details_get_success.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
...
...ystems', 'rating': 700, 'percentile': 61, 'display_url': 'https://dummy.com/app/company/123/diligence-details/?filter=insecure_sys'},<br>{'name': 'Server Software', 'rating': 800, 'percentile': 91, 'display_url': 'https://dummy.com/app/company/123/diligence-details/?filter=server_software'},<br>{'name': 'Desktop Software', 'rating': 500, 'percentile': 9, 'display_url': 'https://dummy.com/app/company/123/diligence-details/?filter=endpoint_pc'},<br>{'name': 'Mobile Software', 'rating': 790, 'percentile': 91, 'display_url': 'https://dummy.com/app/company/123/diligence-details/?filter=endpoint_mobile'},<br>{'name': 'DNSSEC', 'rating': 700, 'percentile': 53, 'display_url': 'https://dummy.com/app/company/123/diligence-details/?filter=dnssec'},<br>{'name': 'Mobile Application Security', 'rating': 'N/A', 'percentile': 'N/A', 'display_url': 'https://dummy.com/app/company/123/diligence-details/?filter=mobile_appsec'},<br>{'name': 'File Sharing', 'rating': 820, 'percentile': 100, 'display_url': 'https://dummy.com/app/company/123/user-behavior'},<br>{'name': 'Security Incidents', 'rating': 820, 'percentile': 100, 'display_url': 'https://dummy.com/app/company/123/rating-details/?vector=news'} |
```
[the remark this answers is not in the document]
Comment (crestdatasystems) [reply Packs/BitSight/Integrations/BitSightForSecurityPerformanceManagement/test_data/company_details_get_success.md:4]: This is not the readme file, rather it is a test file to assert the human-readable of the company details command. This human-readable for the command is the output of CommonServerPython's tableToMarkdown (which internally calls stringEscapeMD) function which implicitly converts new-lines to <br> and not <br/>. The assertion will fail if we replace <br> with <br/> unless tableToMarkdown is updated to use <br/> or before asserting we replace <br> with <br/> in the test case (which does not seem to be a viable option, rather a workaround).

File: Packs/BitSight/Integrations/BitSightForSecurityPerformanceManagement/test_data/company_findings_get_success.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+### Company findings:
+Total Findings: 2
+|Evidence Key|Risk Vector Label|First Seen|Last Seen|ID|Risk Category|Severity|Asset Category|Finding Grade|
+|---|---|---|---|---|---|---|---|---|
+| X.X.X.X:82 | Open Ports | 2021-07-15 | 2021-07-22 | 1001 | Diligence | severe | X.X.X.X:  |  |
+| X.X.X.X:82 | Open Ports | 2021-07-11 | 2021-07-22 | 1002 | Diligence | severe | X.X.X.X: Low<br>X.X.X.Y: Critical | Bad |
```
[the remark this answers is not in the document]
Comment (crestdatasystems) [reply Packs/BitSight/Integrations/BitSightForSecurityPerformanceManagement/test_data/company_findings_get_success.md:6]: Please see the above comment about the same issue.

**Comment by crestdatasystems** [comment]
**Regarding max_fetch** -> We have added the parameter Max Fetch for fetch incidents. Please check again. You will be able to find it being used on line 234 of the script. 

` limit = arg_to_number(args.get('max_fetch', DEFAULT_FETCH_LIMIT), 'Max Fetch', True)`

**Review by MLainer1 — COMMENTED** [review]
Great work :), Let's schedule a demo. 
Make sure to look at the failed checks and fix what you can before the demo.

File: Tests/conf.json [hunk 3]
```diff
@@ -5099,6 +5099,11 @@
             ],
             "playbookID": "GoogleSheets-Test",
             "fromversion": "6.1.0"
+        },
+        {
+            "integrations": "BitSight for Security Performance Management",
+            "playbookID": "BitSight for Security Performance Management-Test",
+            "fromversion": "6.0.0"
         }
```
Comment (MLainer1) [inline Tests/conf.json:5107]:
```suggestion
```
Please remove the test-playbook from the file

Comment (crestdatasystems) [reply Tests/conf.json:5107]: Any specific reason for removing this?

Comment (MLainer1) [reply Tests/conf.json:5107]: We don't add partner test playbook to this file

**Comment by MLainer1** [comment]
@crestdatasystems Please merge from master, There is a validation that fails that may be fixed by the merge

**Comment by MLainer1** [comment]
@crestdatasystems Can you merge again from master? I'm sorry for the inconvenience

**Review by MLainer1 — APPROVED** [review]

**Pull request merged**